src/lyapunov_spectrum.jl

```diff
@@ -31,39 +11,42 @@ together with the components of 4 offset vectors stored in the matrix
 `offset` as columns.
 """=#
 function specular!(p::AbstractParticle{T}, o::Circular{T},
-                   offset::MArray{Tuple{4,4}, T}) where {T<:AbstractFloat}
+                   offset::Vector{SVector{4, T}}) where {T<:AbstractFloat}
     n = normalvec(o, p.pos)
-    ti = [-p.vel[2],p.vel[1]]
+    ti = SV{T}(-p.vel[2],p.vel[1])
+
     cosa = dot(n, -p.vel)
-    p.vel = p.vel - 2*dot(n, p.vel)*n
-    tf = [-p.vel[2], p.vel[1]]
+    p.vel = p.vel + 2*cosa*n
+
+    tf = SV{T}(-p.vel[2], p.vel[1])
+
     for k in 1:4
-        x = offset[k]
+        δΓ = offset[k]
         # Formulas from Dellago, Posch and Hoover, PRE 53, 2, 1996: 1485-1501 (eq. 27)
         # with norm(p) = 1
-        a  = x[δpind] - 2.*dot(x[3:4],n)*n + curvature(o)*2./o.r*dot(x[1:2],ti)/cosa*tf
-        b  = x[1:2] - 2.*dot(x[1:2],n)*n
+        δq  = δΓ[δqind] - 2.*dot(δΓ[δqind],n)*n
+        δp  = δΓ[δpind] - 2.*dot(δΓ[δpind],n)*n - curvature(o)*2./o.r*dot(δΓ[δqind],ti)/cosa*tf
```

Comment:
I am guessing this section will be slightly faster with less accesses. You don't actually use `δΓ`. So maybe it is best to start with `δqprev = offset[k][δqind]` and similarly for `δp`.